### PR TITLE
Added copy method for AuxReaders

### LINKED ISSUE #1785: Add copy method to AuxReader
A copy method needs to be added to `MDAnalysis.auxiliary.base.AuxReader` which should work similarly to the ReaderBase copy method.  This should then be extended to include the [XVGReader](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/auxiliary/XVG.py).

### PR DESCRIPTION
**Pull request by BFedder**
Fixes #1785

Changes made in this Pull Request:
 - Added a copy method for AuxReaders to make them and Universes that have them copyable.
 - Added tests to make sure all currently implemented AuxReaders (XVG, XVG-File, EDR) are copied properly

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/auxiliary/base.py [hunk 1]
```diff
@@ -323,7 +323,9 @@ def __init__(self, represent_ts_as='closest', auxname=None, cutoff=None,
             self.rewind()
 
     def copy(self):
```
Comment (richardjgowers) [inline package/MDAnalysis/auxiliary/base.py:330]: in the docstring can it be made explicit if this is a deep (a completely independent thing with its own file handles etc, no shared resources) or shallow (might share a resource like a file handle with the original) copy?

**Comment by richardjgowers** [comment]
@BFedder this looks good.  With how you're copying the ould it make more sense to define `__getstate__` and `__setstate__` so that `AuxReader` will serialise, then the copying can be done by serialising/deserialising in the same function and returning that?  Two birds with one stone.

**Comment by BFedder** [comment]
I have never used `__getstate__` and `__setstate__` before, so I'm not quite sure... Were you thinking of something like this?

**Review by hmacdope — APPROVED** [review]
This looks good to me @BFedder

File: package/MDAnalysis/auxiliary/base.py [hunk 2]
```diff
@@ -323,7 +324,16 @@ def __init__(self, represent_ts_as='closest', auxname=None, cutoff=None,
...
-        raise NotImplementedError("Copy not implemented for AuxReader")
+        """Returns a deep copy of the AuxReader"""
+        orig_dict = pickle.dumps(self)
+        new_reader = pickle.loads(orig_dict)
+        return new_reader
+
+    def __getstate__(self):
```
Comment (hmacdope) [inline package/MDAnalysis/auxiliary/base.py]: Wait @BFedder quick query does the pickling work without `__getstate__` and `__setstate__` defined here? IIRC pickle does have a default that is very similar to this that does not require the methods to be explicitly defined. 
See [here](https://docs.python.org/3/library/pickle.html#object.__getstate__)

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: package/MDAnalysis/auxiliary/base.py (hunk 1 above)
Comment (IAlibay) [reply package/MDAnalysis/auxiliary/base.py:330]: I'm going to block on this query for docs from @richardjgowers - also changelog, etc.. the usuals

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: package/MDAnalysis/auxiliary/base.py (hunk 1 above)
Comment (hmacdope) [reply package/MDAnalysis/auxiliary/base.py:330]: I think "deep" was meant to clarify

**Comment by hmacdope** [comment]
@IAlibay sorry forgot CHANGELOG.

File: package/MDAnalysis/auxiliary/base.py (hunk 1 above)
Comment (IAlibay) [reply package/MDAnalysis/auxiliary/base.py:330]: Missed that, but please do take the above as an overall "docs" please - versionchanged addition would be good

Comment (hmacdope) [reply package/MDAnalysis/auxiliary/base.py:330]: Yeah sorry I approved too hastily my brain isn't working properly today. 😅

**Comment by BFedder** [comment]
Added versionchanged and changelog entries now. Also, yes @hmacdope, turns out the `__getstate__` and `__setstate__` definitions were redundant - thanks!

**Comment by orbeckst** [comment]
@BFedder would you mind resolving conflicts?

@richardjgowers / @IAlibay can you please look if you're happy or state what needs to be done to complete the PR?

**Review by IAlibay — APPROVED** [review]
I went ahead and fixed the merge issue with the changelog, lgtm!

**Comment by BFedder** [comment]
Thanks for the fix, @IAlibay!

**Review by hmacdope — APPROVED** [review]
LGTM

**Comment by orbeckst** [comment]
@BFedder could you please update the PR and then we can merge it. Sorry for the delay, not sure why we didn't merge more than a month ago. If in doubt, just ping people after a few days...

**Comment by orbeckst** [comment]
darker is complaining about quotation marks ... https://github.com/MDAnalysis/mdanalysis/actions/runs/3859059696/jobs/6578235368 — if you could please fix those, too, then that's one thing less to worry about later

**Comment by BFedder** [comment]
I have taken care of the merge conflicts and the quotation marks darker was complaining about, but now the azure run for Win-Python38-32 bit is failing because `test_write_trajectory_universe` in coordinates/test_h5md.py failed. I'm not sure why this is happening in this setup in particular and nowhere else. Could I please ask for advice on how to fix this?

**Comment by orbeckst** [comment]
https://stackoverflow.com/questions/49151057/valueerror-not-a-location-id-invalid-object-id-while-creating-hdf5-datasets suggests that our error
```
____________ TestH5MDWriterBaseAPI.test_write_trajectory_universe _____________
[gw0] win32 -- Python 3.8.10 C:\hostedtoolcache\windows\Python\3.8.10\x86\python.exe

self = <MDAnalysisTests.coordinates.test_h5md.TestH5MDWriterBaseAPI object at 0x0D0DCF88>
ref = <MDAnalysisTests.coordinates.test_h5md.H5MDReference object at 0x144076D0>
reader = <H5MDReader D:\a\1\s\testsuite\MDAnalysisTests\data\coordinates\test.h5md with 5 frames of 5 atoms>
universe = <Universe with 5 atoms>
tmpdir = local('C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\pytest-of-VssAdministrator\\pytest-0\\popen-gw0\\test_write_trajectory_universe7')

    def test_write_trajectory_universe(self, ref, reader, universe, tmpdir):
        outfile = 'write-uni-test.' + ref.ext
        with tmpdir.as_cwd():
            with ref.writer(outfile, universe.atoms.n_atoms,
                            velocities=True, forces=True) as w:
>               for ts in universe.trajectory:

D:\a\1\s\testsuite\MDAnalysisTests\coordinates\test_h5md.py:135: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\hostedtoolcache\windows\Python\3.8.10\x86\lib\site-packages\mdanalysis-2.5.0.dev0-py3.8-win32.egg\MDAnalysis\coordinates\base.py:795: in __iter__
    self._reopen()
C:\hostedtoolcache\windows\Python\3.8.10\x86\lib\site-packages\mdanalysis-2.5.0.dev0-py3.8-win32.egg\MDAnalysis\coordinates\H5MD.py:747: in _reopen
    self.close()
C:\hostedtoolcache\windows\Python\3.8.10\x86\lib\site-packages\mdanalysis-2.5.0.dev0-py3.8-win32.egg\MDAnalysis\coordinates\H5MD.py:728: in close
    self._file.close()
C:\hostedtoolcache\windows\Python\3.8.10\x86\lib\site-packages\h5py\_hl\files.py:435: in close
    file_list = [x for x in file_list if h5i.get_file_id(x).id == self.id.id]
>   ???
E   ValueError: Not an id of a file object (not an ID of a file object)

h5py\h5i.pyx:114: ValueError
```
is due to access on a closed file.

This seems odd as the same tests pass elsewhere.

I'll try to restart the test (and send a few prayers or curses in the general direction of the code deities).

**Comment by BFedder** [comment]
Your prayers and/or curses appear to have done the trick, thanks! How odd. Is this ready to merge, then?

**Pull request merged**